File: objectModel/CSharp/Microsoft.CommonDataModel.ObjectModel/Persistence/CdmFolder/Types/DataPartition.cs
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="DataPartition.cs" company="Microsoft">
//      All rights reserved.
// </copyright>
// <summary>
//   The representation of data partition in CDM Folders format.
// </summary>
// --------------------------------------------------------------------------------------------------------------------

namespace Microsoft.CommonDataModel.ObjectModel.Persistence.CdmFolder.Types
{
    using System.Collections.Generic;

    using Newtonsoft.Json.Linq;

    /// <summary>
    /// The representation of data partition in CDM Folders format.
    /// </summary>
    public class DataPartition : FileStatus
    {
        /// <summary>
        /// Gets or sets the corpus path for the data file location.
        /// </summary>
        public string Location { get; set; }

        /// <summary>
        /// Gets or sets the exhibited traits.
        /// </summary>
        public List<JToken> ExhibitsTraits { get; set; }

        /// <summary>
        /// Gets or sets the list of key value pairs to give names for the replacement values from the RegEx.
        /// </summary>
        public List<KeyValuePair<string, string>> Arguments { get; set; }

        /// <summary>
        /// Gets or sets the path of a specialized schema to use specifically for the partitions generated.
        /// </summary>
        public string SpecializedSchema { get; set; }

        /// <inheritdoc />
        public string LastFileStatusCheckTime { get; set; }

        /// <inheritdoc />
        public string LastFileModifiedTime { get; set; }

        public string LastChildFileModifiedTime { get; set; }
    }
}

File: objectModel/TypeScript/Persistence/CdmFolder/types/DataPartition.ts
import { FileStatus, TraitReference } from '.';

/**
 * abstract class for a key value pair in typescript.
 */
export abstract class KeyValPair {
    public name: string;
    public value: string;
}

export abstract class DataPartition implements FileStatus {
    /**
     * Gets or sets the name of the data partition.
     */
    public name: string;

    /**
     * The corpus path for the data file location.
     */
    public location: string;

    /**
     * The exhibited traits.
     */
    public exhibitsTraits?: (string | TraitReference)[];

    /**
     * The list of key value pairs to give names for the replacement values from the RegEx.
     */
    public arguments?: KeyValPair[];

    /**
     * The path of a specialized schema to use specifically for the partitions generated.
     */
    public specializedSchema?: string;

    /**
     * Last time the modified times were updated
     */
    public lastFileStatusCheckTime: string;

    /**
     * Last time this file was modified
     */
    public lastFileModifiedTime: string;

    /**
     * Last time the most recently modified child object was modified
     */
    public lastChildFileModifiedTime?: string;
}
